### ISSUE TITLE
i get wrong secret

### ISSUE THREAD

**Issue by tourer-zc** [mraksoll4/yespower_go#1]
i have an encrypted data $y$j9T$nUQ1haLSJODMAlSVUV5NGUw.$tLzBKbQ6ujjnayVUk2/GG1OjhWMqoVJ/05fXYhZH0QD, the real password is `Hillstone!1`. But i can not get the encrypted data with the follwing code:
```go
package main

import (
	"encoding/base64"
	"encoding/hex"
	"fmt"
)

func main() {
	newsalt, _ := base64.RawURLEncoding.DecodeString("nUQ1haLSJODMAlSVUV5NGUw.")
	newpass := []byte("Hillstone!1")

	pass := Yescrypt(newpass, 4096, 32, string(newsalt))
	byt, _ := hex.DecodeString(pass)
	encoded := base64.StdEncoding.EncodeToString(byt)
	fmt.Println(encoded, len(encoded))
}
```

Could you tell me what's wrong in my code?

**Comment by mraksoll4** [comment]
A possible problem is that you are using a modified version for BTE, since it uses a block switch.
https://github.com/mraksoll4/yespower_go/commit/ce59cfa8a0e136adebe7366c346a5cfae66e3cf4

you can try using this code, but there you will need to make some fixes to support 32-bit systems if necessary (compare the codes and use all except the switch.

https://github.com/leviable/yespower-go

**Comment by mraksoll4** [comment]
or remove this funct

![image](https://github.com/mraksoll4/yespower_go/assets/32709596/b7338724-d749-488a-a7fe-85ef7c2d5384)

**Comment by mraksoll4** [comment]
and you can try uncoment this for see what hapening.
![image](https://github.com/mraksoll4/yespower_go/assets/32709596/a5d8d73f-caab-47c1-9b28-9d6965ca0ced)

**Comment by mraksoll4** [comment]
```
package main

import (
	"encoding/base64"
	"encoding/hex"
	"fmt"

	"github.com/mraksoll4/yespower_go"
)

func main() {
	newsalt, _ := base64.RawURLEncoding.DecodeString("nUQ1haLSJODMAlSVUV5NGUw.")
	newpass := []byte("Hillstone!1")

	pass := yespower.Yescrypt(newpass, 4096, 32, string(newsalt))
	byt, _ := hex.DecodeString(pass)
	encoded := base64.StdEncoding.EncodeToString(byt)
	fmt.Println(encoded, len(encoded))
}


```

**Comment by solardiz** [comment]
This is confusing yespower vs. yescrypt. What @mraksoll4 implemented here (thanks!) is yespower 0.5 (similar to yescrypt 0.5, which was a temporary revision during the Password Hashing Competition) and yespower 1.0. What @tourer-zc needs is proper yescrypt 1.0+ as implemented in https://github.com/openwall/yescrypt-go https://pkg.go.dev/github.com/openwall/yescrypt-go, which also has proper parameter decoding and proper base-64 crypt encoding (different from the usual base64).

In other words, what cryptocurrencies call yescrypt (retroactively named yespower 0.5) isn't the same as what we call yescrypt in password security context (yescrypt 1.0 release and on).

**Comment by ghost** [comment]
It is just that you need to remove if else block by time https://github.com/mraksoll4/yespower_go/blob/main/yespower.go#L521.

Otherwise, this library will reproduce same results as the original C library of it

https://github.com/cpuchain/go-yespower/blob/main/yespower_test.go

For comparison I have added test cases generated with random 32 sized bytes.

**Comment by solardiz** [comment]
> It is just that you need to remove if else block by time https://github.com/mraksoll4/yespower_go/blob/main/yespower.go#L521.
> 
> Otherwise, this library will reproduce same results as the original C library of it

That's cool if so, but it wouldn't help @tourer-zc who wanted yescrypt 1.0 (as modern Linux distros use for password hashing) and _not_ yespower. By "the original C library of it", you mean yespower, not yescrypt. That "if else block by time" you reference is switching between yespower 0.5 (which it calls "yescrypt" for historical reasons) and 1.0.
